### ISSUE TITLE
[Bug] AccessDeniedException on CreatePortfolio

### ISSUE THREAD

**Issue by gazzyt** [awslabs/service-workbench-on-aws#888]
**Describe the bug**
`environment-deploy.sh` script terminates during post deployment due to the following error in the post deployment lambda:

```
{
    "errorType": "AccessDeniedException",
    "errorMessage": "User: arn:aws:sts::XXXXXXXXXXXX:assumed-role/tst-irl-sw-t-postDeployme-RolePostDeploymentLambda-QC293SN0CGTH/irl-sw-t-postDeployment-tst-postDeployment is not authorized to perform: servicecatalog:CreatePortfolio on resource: arn:aws:catalog:eu-west-1:XXXXXXXXXXXX:* because no identity-based policy allows the servicecatalog:CreatePortfolio action",
    
	"trace": [
        "AccessDeniedException: User: arn:aws:sts::XXXXXXXXXXXX:assumed-role/tst-irl-sw-t-postDeployme-RolePostDeploymentLambda-QC293SN0CGTH/irl-sw-t-postDeployment-tst-postDeployment is not authorized to perform: servicecatalog:CreatePortfolio on resource: arn:aws:catalog:eu-west-1:XXXXXXXXXXXX:* because no identity-based policy allows the servicecatalog:CreatePortfolio action",
        "    at Request.extractError (/var/runtime/node_modules/aws-sdk/lib/protocol/json.js:52:27)",
        "    at Request.callListeners (/var/runtime/node_modules/aws-sdk/lib/sequential_executor.js:106:20)",
        "    at Request.emit (/var/runtime/node_modules/aws-sdk/lib/sequential_executor.js:78:10)",
        "    at Request.emit (/var/runtime/node_modules/aws-sdk/lib/request.js:688:14)",
        "    at Request.transition (/var/runtime/node_modules/aws-sdk/lib/request.js:22:10)",
        "    at AcceptorStateMachine.runTo (/var/runtime/node_modules/aws-sdk/lib/state_machine.js:14:12)",
        "    at /var/runtime/node_modules/aws-sdk/lib/state_machine.js:26:10",
        "    at Request.<anonymous> (/var/runtime/node_modules/aws-sdk/lib/request.js:38:9)",
        "    at Request.<anonymous> (/var/runtime/node_modules/aws-sdk/lib/request.js:690:12)",
        "    at Request.callListeners (/var/runtime/node_modules/aws-sdk/lib/sequential_executor.js:116:18)"
    ]
}
--------------------------------------------------------------------
var/runtime/node_modules/aws-sdk/lib/request.js:38:9)\n    at Request.<anonymous> (/var/runtime/node_modules/aws-sdk/lib/request.js:690:12)\n    at Request.callListeners (/var/runtime/node_modules/aws-sdk/lib/sequential_executor.js:116:18)",
  "message": "User: arn:aws:sts::XXXXXXXXXXXX:assumed-role/tst-irl-sw-t-postDeployme-RolePostDeploymentLambda-QC293SN0CGTH/irl-sw-t-postDeployment-tst-postDeployment is not authorized to perform: servicecatalog:CreatePortfolio on resource: arn:aws:catalog:eu-west-1:XXXXXXXXXXXX:* because no identity-based policy allows the servicecatalog:CreatePortfolio action",
  "code": "AccessDeniedException",
  "time": "2022-01-20T12:39:42.056Z",
  "requestId": "c3ff62fb-f209-401c-bbb4-760f146d7a20",
  "statusCode": 400,
  "retryable": false,
  "retryDelay": 813.9565638827113
}
```

I was able to work around this by simply adding that resource to the policy in `main/solution/post-deployment/config/infra/cloudformation.yml` as shown in the snippet below, after which the installation ran successfully.
```
        - PolicyName: service-catalog-access
          PolicyDocument:
            Statement:
              Effect: Allow
              Action:
                - servicecatalog:CreatePortfolio
                - servicecatalog:UpdatePortfolio
                - servicecatalog:DescribePortfolio
                - servicecatalog:CreateProduct
                - servicecatalog:UpdateProduct
                - servicecatalog:DescribeProduct
                - servicecatalog:AssociatePrincipalWithPortfolio
                - servicecatalog:AssociateProductWithPortfolio
                - servicecatalog:CreateConstraint
                - servicecatalog:ListPortfolios
                - servicecatalog:CreateProvisioningArtifact
              Resource:
                - !Sub arn:${AWS::Partition}:servicecatalog:${AWS::Region}:${AWS::AccountId}:*
                - !Sub arn:${AWS::Partition}:servicecatalog:${AWS::Region}:${AWS::AccountId}:*/*
                - !Sub arn:${AWS::Partition}:catalog:${AWS::Region}:${AWS::AccountId}:*
                - !Sub arn:${AWS::Partition}:catalog:${AWS::Region}:${AWS::AccountId}:portfolio/*
                - !Sub arn:${AWS::Partition}:catalog:${AWS::Region}:${AWS::AccountId}:product/*
```

While this workaround does work I am suspicious that after some googling I can't find anyone else who is running into this problem and the resource I added is not documented as required by AWS (the existing 4 are). Also this lambda is running in the AWS cloud (with the AWS SDK they provide) so there seems little scope for environmental issues in my dev environment.

**To Reproduce**
1. Run new deployment via `./scripts/environment-deploy.sh ${STAGE_NAME}`
2. Script terminates during post-deployment steps
3. Service Catalog Portfolio has not been created (verified in AWS console)

**Expected behavior**
Script runs successfully to completion and Service Catalog Portfolio can be seen in AWS console

**Versions (please complete the following information):**
 - Release Version installed from v4.2.0 tag
 - Also tried mainline with same results

**Additional context**

**Comment by Bingjiling** [comment]
Hi @gazzyt,

Thanks for reaching out! This issue was introduced by Service Catalog recent update. A temporary fix has been merged in mainline through [this PR](https://github.com/awslabs/service-workbench-on-aws/pull/878). We'll be releasing this patch shortly. 

Thanks,
Yanyu

**Comment by Bingjiling** [comment]
Hi @gazzyt,

The patch for this issue is released in [v4.3.0](https://github.com/awslabs/service-workbench-on-aws/releases/tag/crea). It contains the same fix you described. Thanks again for reaching out!

Thanks,
Yanyu
